test/integration/smoke/test_vm_life_cycle.py

```diff
@@ -1011,8 +1011,36 @@ def setUpClass(cls):
 
     @classmethod
     def tearDownClass(cls):
+        cls.ensure_all_hosts_are_up()
         super(TestSecuredVmMigration, cls).tearDownClass()
 
+    @classmethod
+    def ensure_all_hosts_are_up(cls):
+        hosts = Host.list(
+            cls.apiclient,
+            zoneid=cls.zone.id,
+            type='Routing',
+            hypervisor='KVM'
+        )
+        for host in hosts:
+            if host.state != "Up":
+                SshClient(host.ipaddress, port=22, user=cls.hostConfig["username"], passwd=cls.hostConfig["password"]) \
+                    .execute("service cloudstack-agent stop ; \
+                              sleep 10 ; \
+                              service cloudstack-agent start")
+                interval = 5
+                retries = 10
+                while retries > -1:
+                    time.sleep(interval)
+                    restarted_host = Host.list(
+                        cls.apiclient,
+                        hostid=host.id,
+                        type='Routing'
+                    )[0]
+                    if restarted_host.state == state:
```

Comment:
should `== state` be `== "Up"` ?

Reply:
@weizhouapache yes 🙈
But I'm not able to reproduce the issue with hosts remaining in `Connecting` state so not sure if this will solve the problem